### PR TITLE
Improve ci

### PR DESCRIPTION
**Pull request by Emilgardis**
This pr includes

- Support for the changes done in v0.13
- `--toolchain foo` to specify what toolchain to build the chips with (makes building locally easier, no need to rebuild `svd2rust-regress`)
- Add support for `--chip XXX YYY`
- Added validators to chip, manufacturer and architecture
- `-vv` will now output all stderr logs
- Added `--list` to quickly troubleshoot what test cases will be run
- A patched ATSAMD21G18A svd, thanks to @wez 

## Open "issues"
- Should we ignore case when filtering?
- Should a `--chip XXX` always run even if it is set to only run on `--long-test` (and possibly `--bad-test`)
- Add a `--` command to pass arguments to `svd2rust` and possibly also a way to pass arguments to `cargo check`
- Colorize output
- Add regress:beta test to gitlab-runner
- Move test cases to a or multiple config file(s) (e.g json/toml/yaml)

cc @jamesmunns

### CODE REVIEW

**Comment by jamesmunns** [comment]
Hey @Emilgardis, ACKing this as seen, looks good overall, will have more time tomorrow (2018-05-21) to look at this.

I'd add one open issue: IMO It probably makes more sense to move all the test cases to one or more json/toml/yaml files rather than keeping it in code.

File: ci/svd2rust-regress/src/svd_test.rs [hunk 1]
```diff
@@ -1,16 +1,19 @@
...
 use reqwest;
-use std::fs::{remove_dir_all, File, OpenOptions};
+use std::fs::{self, File, OpenOptions};
 use std::io::prelude::*;
 use std::path::PathBuf;
 use std::process::{Command, Output};
 use tests::TestCase;
 
-static CRATES_ALL: &[&str] = &["bare-metal = \"0.1.0\"", "vcell = \"0.1.0\""];
+static CRATES_ALL: &[&str] = &["bare-metal = \"0.2.0\"", "vcell = \"0.1.0\""];
```
Comment (jamesmunns) [inline ci/svd2rust-regress/src/svd_test.rs]: @Emilgardis is this info stored anywhere in `svd2rust` that we could use to prevent dual maintenance here?

**Comment by jamesmunns** [comment]
Overall LGTM, if we can dedupe the dependent crates somehow, that would be nice, also feel free to open up one or more follow-up issues for the `todos` if you would like to go ahead and merge this.

File: ci/svd2rust-regress/src/tests.rs [hunk 2]
```diff
@@ -706,9 +706,9 @@ pub const TESTS: &'static [&'static TestCase] = &[
         arch: CortexM,
         mfgr: Atmel,
         chip: "ATSAMD21G18A",
-        svd_url: None,
-        should_pass: false,
-        run_when: Never,
+        svd_url: Some("https://raw.githubusercontent.com/wez/atsamd21-rs/master/svd/ATSAMD21G18A.svd"),
```
Comment (Emilgardis) [inline ci/svd2rust-regress/src/tests.rs]: This should be a specific commit

**Comment by mathk** [comment]
bors try

**Comment by mathk** [comment]
bors try

**Comment by mathk** [comment]
Ping from triage: @Emilgardis can you rebase/merge.

**Comment by Disasm** [comment]
@jamesmunns what is the current status of this PR? Does it need some discussion/changes or can it be merged right now?

**Comment by therealprof** [comment]
bors try

**Pull request merged**